multitenancy-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenantEventSchedulerTest.java

```diff
@@ -0,0 +1,157 @@
+
+import org.axonframework.common.Registration;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.eventhandling.scheduling.ScheduleToken;
+import org.axonframework.extensions.multitenancy.components.TargetTenantResolver;
+import org.axonframework.extensions.multitenancy.components.TenantDescriptor;
+import org.axonframework.extensions.multitenancy.components.eventstore.TenantEventSchedulerSegmentFactory;
+import org.junit.jupiter.api.*;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Stefan Dragisic
+ */
+class MultiTenantEventSchedulerTest {
+
+    private MultiTenantEventScheduler testSubject;
+    private EventScheduler fixtureSegment1;
+    private EventScheduler fixtureSegment2;
+
+    @BeforeEach
+    void setUp() {
+        fixtureSegment1 = mock(EventScheduler.class);
+        fixtureSegment2 = mock(EventScheduler.class);
+
+        TenantEventSchedulerSegmentFactory tenantSegmentFactory = t -> {
+            if (t.tenantId().equals("fixtureTenant1")) {
+                return fixtureSegment1;
+            } else {
+                return fixtureSegment2;
+            }
+        };
+        TargetTenantResolver<EventMessage<?>> targetTenantResolver = (m, tenants) -> TenantDescriptor.tenantWithId(
+                "fixtureTenant2");
+
+        testSubject = MultiTenantEventScheduler.builder()
+                                           .tenantSegmentFactory(tenantSegmentFactory)
+                                           .targetTenantResolver(targetTenantResolver)
+                                           .build();
+    }
+    @Test
+    void scheduleWithInstant() {
+        testSubject.registerTenant(TenantDescriptor.tenantWithId("fixtureTenant1"));
+        testSubject.registerTenant(TenantDescriptor.tenantWithId("fixtureTenant2"));
+
+        EventMessage<Object> event = GenericEventMessage.asEventMessage("event");
+        Instant instant = Instant.MAX;
+        testSubject.schedule(instant, event);
+        verify(fixtureSegment2).schedule(instant, event);
+        verify(fixtureSegment1, times(0)).schedule(any(Instant.class), any());
+    }
+
+
+    @Test
+    void scheduleWithDuration() {
+        testSubject.registerTenant(TenantDescriptor.tenantWithId("fixtureTenant1"));
+        testSubject.registerTenant(TenantDescriptor.tenantWithId("fixtureTenant2"));
+
+        EventMessage<Object> event = GenericEventMessage.asEventMessage("event");
+        Duration duration = Duration.ZERO;
+        testSubject.schedule(duration, event);
+        verify(fixtureSegment2).schedule(duration, event);
+        verify(fixtureSegment1, times(0)).schedule(any(Duration.class), any());
+    }
+
+    @Test
+    void scheduleObject() {
```

Comment:
Nit: I would rename the test to clarify this should fail *because* there is no `EventMessage` and `TenantDescriptor`. Actually, I would test both these cases.